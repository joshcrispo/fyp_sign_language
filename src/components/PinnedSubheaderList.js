import * as React from 'react';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';
import ListSubheader from '@mui/material/ListSubheader';

export default function PinnedSubheaderList({ models, selectedModel, onModelSelect }) {
  return (
    <List
    sx={{
      bgcolor: 'rgba(0, 0, 0, 0.2)',
      color: '#f3f3d6',
      position: 'relative',
      overflow: 'auto',
      maxHeight: 450,
      marginBottom: 2,
      borderRadius: '16px',
      boxShadow: '0 0 10px rgba(0, 0, 0, 0.2)',
      '& ul': { padding: 0 },
      '& .MuiListSubheader-root': {
        bgcolor: '#f3f3d6',
        color: 'rgba(0, 0, 0, 0.87)',
        borderRadius: '16px 16px 0 0',
      },
      '& .MuiListItem-root': {
        '&:hover': {
          bgcolor: 'rgba(255, 255, 255, 0.2)',
        }
      }
    }}
      subheader={<li />}
    >
      <li key="section-models">
        <ul>
          <ListSubheader>{`Models`}</ListSubheader>
          {models.map((model, index) => (
            <ListItem key={`item-model-${index}`} disablePadding>
              <ListItemButton
                onClick={() => onModelSelect(model)}
                sx={{
                  bgcolor: model === selectedModel ? 'rgba(255, 255, 255, 0.12)' : 'inherit',
                  '&:hover': {
                    bgcolor: 'rgba(255, 255, 255, 0.2)',
                  },
                }}
              >
                <ListItemText primary={model} />
              </ListItemButton>
            </ListItem>
          ))}
        </ul>
      </li>
    </List>
  );
}