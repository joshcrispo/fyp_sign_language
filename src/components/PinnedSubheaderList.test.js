import { fireEvent, render, screen } from '@testing-library/react';
import PinnedSubheaderList from './PinnedSubheaderList';

test('renders "Getting Started" model in the list', () => {
    const models = ["Getting Started", "Another Model"];
    const onModelSelect = jest.fn();

    render(<PinnedSubheaderList models={models} selectedModel={"Getting Started"} onModelSelect={onModelSelect} />);
    const listItemElement = screen.getByText(/another model/i);
    expect(listItemElement).toBeInTheDocument();
  });

test('renders "Another" model in the list', () => {
    const models = ["Getting Started", "Another Model"];
    const onModelSelect = jest.fn();

    render(<PinnedSubheaderList models={models} selectedModel={"Getting Started"} onModelSelect={onModelSelect} />);
    const listItemElement = screen.getByText(/another model/i);
    expect(listItemElement).toBeInTheDocument();
});

test('doesntrenders model not in the list', () => {
    const models = ["Getting Started", "Another Model"];
    const onModelSelect = jest.fn();
  
    render(<PinnedSubheaderList models={models} selectedModel={"Getting Started"} onModelSelect={onModelSelect} />);
    const listItemElement = screen.queryByText(/no model/i);
    expect(listItemElement).not.toBeInTheDocument();
});

test('PinnedSubheaderList renders correctly and matches snapshot', () => {
    const models = ["Getting Started", "Another Model"];
    const { asFragment } = render(<PinnedSubheaderList models={models} selectedModel="Getting Started" onModelSelect={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
});

test('selecting a model triggers onModelSelect callback', () => {
    const models = ["Getting Started", "Another Model"];
    const onModelSelect = jest.fn();
    render(<PinnedSubheaderList models={models} selectedModel={"Another Model"} onModelSelect={onModelSelect} />);
    
    const modelButton = screen.getByRole('button', { name: /another model/i });
    fireEvent.click(modelButton);
    
    expect(onModelSelect).toHaveBeenCalledWith("Another Model");
});
