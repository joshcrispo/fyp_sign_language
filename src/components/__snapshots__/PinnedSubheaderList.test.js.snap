// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinnedSubheaderList renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding MuiList-subheader css-zlnxci-MuiList-root"
  >
    <li />
    <li>
      <ul>
        <li
          class="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-uob957-MuiListSubheader-root"
        >
          Categories
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-4f85g-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Getting Started
              </span>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-2ig67u-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Another Model
              </span>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </li>
      </ul>
    </li>
  </ul>
</DocumentFragment>
`;
